### PR TITLE
[inference][trt] update roll op  2 gather layer

### PR DESCRIPTION
**Pull request by bukejiyu**
### PR types
Bug fixes

### PR changes
OPs

### Description
- update roll for TRT, use Gather layer

### CODE REVIEW

**Review by zhangjun — DISMISSED** [review]
LGTM

File: paddle/fluid/inference/tensorrt/convert/roll_op.cc [hunk 1]
```diff
@@ -30,46 +30,63 @@ class RollOpConverter : public OpConverter {
```
Comment (zhoutianzi666) [inline paddle/fluid/inference/tensorrt/convert/roll_op.cc:30]: 改成Gather layer

**Review by zhoutianzi666 — APPROVED** [review]
LGTM

**Review by zhangjun — APPROVED** [review]
LGTM

**Pull request merged**